src/Core/Services/Implementations/PostalMailDeliveryService.cs

```diff
@@ -0,0 +1,115 @@
+﻿using System;
+using System.Threading.Tasks;
+using Bit.Core.Settings;
+using Microsoft.Extensions.Logging;
+using System.Net.Http;
+using System.Collections.Generic;
+using Newtonsoft.Json;
+using Microsoft.AspNetCore.Hosting;
+using System.Text;
+
+namespace Bit.Core.Services
+{
+    public class PostalMailDeliveryService : IMailDeliveryService
+    {
+        private readonly GlobalSettings _globalSettings;
+        private readonly ILogger<PostalMailDeliveryService> _logger;
+        private readonly IHttpClientFactory _clientFactory;
+        private readonly string _baseTag;
+        private readonly string _from;
+        private readonly string _reply;
+
+        public PostalMailDeliveryService(
+            GlobalSettings globalSettings,
+            ILogger<PostalMailDeliveryService> logger,
+            IWebHostEnvironment hostingEnvironment,
+            IHttpClientFactory clientFactory)
+        {
+            _globalSettings = globalSettings;
+            _logger = logger;
+            _clientFactory = clientFactory;
+            _baseTag = $"Env_{hostingEnvironment.EnvironmentName}-" +
+                $"Server_{globalSettings.ProjectName?.Replace(' ', '_')}";
+            _from = $"\"{globalSettings.SiteName}\" <no-reply@{_globalSettings.Mail.PostalDomain}>";
```

Comment:
any thoughts on making the entire address configurable like `{_globalSettings.Mail.ReplyToEmail}`; I didn't see in the other providers using `no-reply@` explicitly, not sure if this is desired to deviate from the `Reply-To` header, or necessary.

Reply:
It has to be different than the reply-to since it has to come from a specific domain. We could make it configurable, but it would just be another global setting properly. We need PostalDomain for the API, so I was just trying to reuse it before creating another property.

Reply:
sounds good for now, thanks.